### PR TITLE
document front_controller_active

### LINKED ISSUE #12345: Document front_controller_active in config.sample.php
From https://github.com/owncloud/core/issues/8269#issuecomment-63934300

>    @DeepDiver1975 I checked the sample file on my 7.0.3 installation and https://github.com/owncloud/core/blob/master/config/config.sample.php in the repo. I couldn't find any reference to index.php.

Looks like the config flag is not documented - sorry for this. The config setting is named 'front_controller_active' and it should be set to true.

### PR DESCRIPTION
**Pull request by MorrisJobke**
cc @DeepDiver1975 @LukasReschke 

I need your help regarding the webserver config. Because we have apps/files/index.php the simple FallbackResource doesn't trigger very well. I guess we need to adjust RewriteRules.

fixes #12345 once it is done

### CODE REVIEW

**Comment by nickvergessen** [comment]
Failing avatar tests :see_no_evil:

**Comment by MorrisJobke** [comment]
fixed on master

**Comment by MorrisJobke** [comment]
@owncloud-bot retest this please

File: .htaccess [hunk 1]
```diff
@@ -38,6 +38,7 @@ AddEncoding gzip svgz
 </IfModule>
 <IfModule mod_dir.c>
 DirectoryIndex index.php index.html
+FallbackResource index.php
```
Comment (DeepDiver1975) [inline .htaccess]: I'm against this change - I tried to get owncloud properly running using FallbackResource and this did not really work out.

File: config/config.sample.php [hunk 2]
```diff
@@ -985,4 +985,11 @@
  */
 'copied_sample_config' => true,
 
+/**
+ * If this option is set to 'true' all URL will be generated without the
+ * index.php. Keep in mind to setup your webserver to still match the URLs
+ * correct.
+ */
+'front_controller_active' => 'false'
```
Comment (DeepDiver1975) [inline config/config.sample.php]: I'm not 100% sire if we should document this yet - I basically started out with this - but from my understanding we are not yet fully there.

But maybe I miss something ....

File: .htaccess (hunk 1 above)
Comment (LukasReschke) [inline .htaccess]: Yes - that will very likely cause problems. We really need some blackmagic such as https://github.com/owncloud/core/pull/14081/files#diff-8052c42ab3b8aa06a3f5f788a4ddccc2R52

**Comment by DeepDiver1975** [comment]
refs https://github.com/owncloud/core/pull/14081

File: .htaccess (hunk 1 above)
Comment (MorrisJobke) [inline .htaccess]: Oh sorry ... I didn't want to commit that initially ... this should just show what I tried.

**Comment by DeepDiver1975** [comment]
> refs #14081

@MorrisJobke I'd rather go for #14081

**Pull request closed without merging**